java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaObjectOutputStream.java

```diff
@@ -170,54 +177,63 @@ public Thread newThread(final Runnable runnable) {
          * We have to wait here until the upload to Manta starts and a Writer
          * becomes available.
          */
-        while (httpContent.getWriter() == null) {
-            try {
-                Thread.sleep(CLOSED_CHECK_INTERVAL);
-            } catch (InterruptedException e) {
-                return;
+        synchronized (this.httpContent) {
+            while (!isClosed() && !httpContent.isClosed() && httpContent.getWriter() == null) {
```

Comment:
Is `httpContent` the  `innerClosed` below?  I'm having trouble seeing the case where only one of these is closed.